src/main/java/programminglife/model/drawing/DrawableSNP.java

```diff
@@ -27,16 +27,12 @@
     DrawableSNP(DrawableNode parent, DrawableNode child, Collection<DrawableNode> mutations) {
         super(mutations.iterator().next().getGraph(), -mutations.hashCode());
 
-        if (!(parent instanceof DrawableSegment)) {
-            throw new IllegalArgumentException("Parent of SNP must be a Segment!");
-        } else if (!(child instanceof DrawableSegment)) {
-            throw new IllegalArgumentException("Child of SNP must be a Segment!");
-        } else if (!mutations.stream().allMatch(DrawableSegment.class::isInstance)) {
+        if (!mutations.stream().allMatch(DrawableSegment.class::isInstance)) {
```

Comment:
This is not the responsibility of the DrawableSNP, it is the responsibility of the caller (so make `mutations` a `Collection<DrawableSegment>`)